packages/autocomplete-shared/src/core/AutocompleteOptions.ts

```diff
@@ -11,6 +11,15 @@ import {
 } from './AutocompleteSource';
 import { AutocompleteState } from './AutocompleteState';
 
+export type AutocompleteEnterKeyHint =
+  | 'enter'
+  | 'done'
+  | 'go'
+  | 'next'
+  | 'previous'
+  | 'search'
+  | 'send';
```

Comment:
Sadly `HTMLElement['enterKeyHint']` is too broad, so I went for an explicit listing of allowed values, based on https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/enterkeyhint#values.